app/controllers/programs_controller.rb

```diff
@@ -19,6 +19,7 @@ def index
   end
 
   def duplicate
+    @copy_from_id = @program.id
```

Comment:
Please consider renaming this instance variable so its easier to understand. May be _duplicate_program_id_